### ISSUE TITLE
add option to rename files in borg recreate

### ISSUE THREAD

**Issue by cyril42e** [borgbackup/borg#1857]
I have a suggestion, but I don't know how common is this need. I created archives with different paths prefixes (sometimes absolute, sometimes relative), and when I will recreate them (for multiple reasons), I would like also to change the path of some of them to make all the archives homogeneous (cleaner when mounted, and allows deduplication of metadata).

It could be an option `--remove-prefix`, and maybe an additional option `--add-prefix`, or instead a more general option `--rename` that takes a search-and-replace regex ("s|^/mnt/||" to remove a prefix, "s|^|/mnt/|" to add a prefix, "s|^data/|Data/|" to modify a prefix, "s|/travail/|/work/|" to modify a part of the path, ...).

Does it sound useful enough ?

**Comment by RonnyPfannschmidt** [comment]
is this potentially a duplicate of #1650 (all-tho the regex renaming is more nuanced)

**Comment by cyril42e** [comment]
Indeed it seems that the `--remove-prefix` I suggested here is equivalent to the `--strip-components` you suggested in #1650. Is it the part that was estimated to be to complex ? On another hand, the fact that it exists in `tar` is a good point for its utility, and it could be implemented for `borg extract` as well. The `--only` that you also suggested in #1650 is already handled by the `PATH` argument of `borg recreate`, right ?

**Comment by RonnyPfannschmidt** [comment]
@cyril42e correct

**Comment by billop** [comment]
Relationship with #1650 is still confusing to me. 

Anyways, I just want to say that this would be very useful for me since I started off a long time ago backing up my whole system, but these days I've realized that with proper system config management, I only really want to back up my home directory. However I'm stuck backing up `/home/x/{Documents,Music,etc.}` rather than directly backing up `Documents, Music, etc.`

**Comment by ThomasWaldmann** [comment]
Why are you "stuck" with that? You can change it any time...

**Comment by billop** [comment]
> Why are you "stuck" with that? You can change it any time...

What do you mean? I thought that moving paths in recreate wasn't possible.

**Comment by ThomasWaldmann** [comment]
What I meant is to change the `borg create` command and just have the desired path layout for future archives.

**Comment by billop** [comment]
@ThomasWaldmann would deduplication still work? That was my main concern.

**Comment by ThomasWaldmann** [comment]
@billop yes, deduplication in borg works based on file **content**, not on file **name** (path). to be precise, as long as the chunker produces the same chunks, it will work fine.

BUT (not relevant here I guess), the first backup after you change absolute pathes in the filesystem, borg will be slower due to misses in the files cache.

**Comment by billop** [comment]
@ThomasWaldmann wow that's amazing. Thanks.

**Comment by ThomasWaldmann** [comment]
I am closing this, because besides maybe #6461, there is no plan to implement "renaming" in `borg recreate`.
